search/ranking_utils.hpp

```diff
@@ -17,9 +17,36 @@
 #include <string>
 #include <vector>
 
+class CategoriesHolder;
+
+namespace feature
+{
+class TypesHolder;
+}
+
 namespace search
 {
 class QueryParams;
+class TokenSlice;
+
+class CategoriesInfo
+{
+public:
+  CategoriesInfo(feature::TypesHolder const & holder, TokenSlice const & tokens,
+                 Locales const & locales, CategoriesHolder const & categories);
+
+  // Returns true when all tokens correspond to categories in
+  // |holder|.
```

Comment:
It's not clear what |holder| is here. Probably add a local copy, |m_holder|?

Reply:
I'd prefer not to hold reference to TypesHolder.